### ISSUE TITLE
How are you supposed to build it

### ISSUE THREAD

**Issue by xiaoxiao921** [ncatlin/exileSniffer#4]
Hi, i'm really interested on this project sadly there is no indication on how to even build this project.
I have installed most libs required and the only part missing is the QT part, no trace of .pro files or anything, obviously Visual Studio tells me about how he can't #include <QtWidgets> and things like that :(

**Comment by leethobbit** [comment]
I was going to post on here but have been working through it.  I have solved the compiling errors and am working on linker errors (probably caused by me moving libs around or missing some still) but I can post an update when I get it fully compiling.

**Comment by leethobbit** [comment]
Yeah, nevermind lol.  I got it to compile in VS2017 without issues now and it creates the exe and a pdb with linker file but it is missing all of the GUI stuff and dlls.  I tried moving all of my files from her compiled version into the folder and it shows a procedure entry point error when I try to run it.  I am going to continue playing around with this but I'm not sure I can get it working.  I might just have to rewrite it and copy over the core functions.

**Comment by ncatlin** [comment]
Glad you got it compiling! I was mystified about the ui thing since all the files are there. If it helps, the version of Qt i'm compiling it with is 5.12.0\msvc2017_64. QAbstractItemView is a standard Qt thing (https://doc.qt.io/qt-5/qabstractitemview.html).

**Comment by leethobbit** [comment]
May I ask how you are compiling exileSniffer?  I am still confused about if you can compile the program in Qt Creator or if Qt is strictly for UI design.  I am currently using Visual Studio 2017 Community with the Qt VS Tools but if this is not the proper way it could explain my problems.  Thanks so much for the reply!

**Comment by ncatlin** [comment]
You've got it right - i've only used the Qt tools for design, Visual Studio 2017 Community for all the building. It sounds like you are really close but linker errors are a big pain to debug.

**Comment by leethobbit** [comment]
![image](https://user-images.githubusercontent.com/47875749/54781259-29fb1380-4bf2-11e9-9b82-40ec636ed65f.png)

That's the error when I try to run the application after creating it.  I will continue to work on it and don't expect you to spend time debugging but I wanted to post it just in case you've seen it before and might know a quick fix.  Thanks again for sharing such an awesome tool, I have plans for getting it working for the current and future leagues :)

**Comment by ncatlin** [comment]
Be my guest :)

**Comment by ncatlin** [comment]
I've also pushed a few old code changes that I didn't commit before, obviously it won't make it compatible with the latest servers but adds a few more packets, supports the steam client and might fix a crash or typo or two.

**Comment by leethobbit** [comment]
Thanks for sharing new commits!  The new version is yelling at me about not being able to access the a protected member in CryptoPP::Salsa20_Policy - I know this is part of the modified headers you shared and I verified that the salsa.h and strciphr.h both have your modifications to make that m_state variable public.  I don't think I had this issue with the original commits I used.

**Comment by leethobbit** [comment]
HEY!  These new commits are working better.  exileSniffer no longer crashes but does complain of a bunch of badly decoded packets.  I really, REALLY appreciate you pushing these commits because I have a much better idea of what to do now.  Thank you so much, I'll be submitting a pull request probably to the readme file with more detailed instructions and troubleshooting for compiling.  Thanks again you are a true hero!

**Comment by ncatlin** [comment]
That's fantastic to hear, glad it's still working with the live servers!
